tamr_unify_client/models/dataset/resource.py

```diff
@@ -65,21 +67,54 @@ def update_records(self, records, **json_args):
         :returns: JSON response body from server.
         :rtype: :py:class:`dict`
         """
-
-        def _stringify_updates(updates):
-            for update in updates:
-                yield json.dumps(update, **json_args).encode("utf-8")
+        stringified_updates = (
+            json.dumps(update, **json_args).encode("utf-8") for update in updates
+        )
 
         return (
             self.client.post(
                 self.api_path + ":updateRecords",
                 headers={"Content-Encoding": "utf-8"},
-                data=_stringify_updates(records),
+                data=stringified_updates,
             )
             .successful()
             .json()
         )
 
+    def upsert_records(self, records, primary_key_name, **json_args):
+        """Creates or updates the specified records.
+
+        :param records: The records to update, as dictionaries.
+        :type records: iterable[dict]
+        :param primary_key_name: The name of the primary key for these records, which must be a key in each record dictionary.
+        :type primary_key_name: str
+        :param `**json_args`: Arguments to pass to the JSON `dumps` function, as documented `here <https://simplejson.readthedocs.io/en/latest/#simplejson.dumps>`_.
+            Some of these, such as `indent`, may not work with Unify.
+        :return: JSON response body from the server.
+        :rtype: dict
+        """
+        updates = (
+            {"action": "CREATE", "recordId": record[primary_key_name], "record": record}
+            for record in records
+        )
+        return self._update_records(updates, **json_args)
+
+    def delete_records(self, records, primary_key_name):
```

Comment:
There's another use-case where I want to delete a record, but I only know its ID. In this case, would be nice to have a `dataset.delete_records_by_id(record_ids)`. @juliamcclellan that can be part of this PR or we can make an issue for it and do it later.